pyo3-introspection/src/introspection.rs

```diff
@@ -311,85 +311,40 @@ fn find_introspection_chunks_in_macho(
                 &library_content[usize::try_from(data_offset).context("File offset overflow")?..],
-                0,
-                library_content,
-                macho.is_64,
             )?);
         }
     }
     Ok(chunks)
 }
 
 fn find_introspection_chunks_in_pe(pe: &PE<'_>, library_content: &[u8]) -> Result<Vec<Chunk>> {
-    let rdata_data_section = pe
-        .sections
-        .iter()
-        .find(|section| section.name().unwrap_or_default() == ".rdata")
-        .context("No .rdata section found")?;
-    let rdata_shift = usize::try_from(pe.image_base).context("image_base overflow")?
-        + usize::try_from(rdata_data_section.virtual_address)
-            .context(".rdata virtual_address overflow")?
-        - usize::try_from(rdata_data_section.pointer_to_raw_data)
-            .context(".rdata pointer_to_raw_data overflow")?;
-
     let mut chunks = Vec::new();
     for export in &pe.exports {
         if is_introspection_symbol(export.name.unwrap_or_default()) {
-            chunks.push(read_symbol_value_with_ptr_and_len(
+            chunks.push(deserialize_prefix(
                 &library_content[export.offset.context("No symbol offset")?..],
-                rdata_shift,
-                library_content,
-                pe.is_64,
             )?);
         }
     }
     Ok(chunks)
 }
 
-fn read_symbol_value_with_ptr_and_len(
-    value_slice: &[u8],
-    shift: usize,
-    full_library_content: &[u8],
-    is_64: bool,
-) -> Result<Chunk> {
-    let (ptr, len) = if is_64 {
-        let (ptr, len) = value_slice[..16].split_at(8);
-        let ptr = usize::try_from(u64::from_le_bytes(
-            ptr.try_into().context("Too short symbol value")?,
-        ))
-        .context("Pointer overflow")?;
-        let len = usize::try_from(u64::from_le_bytes(
-            len.try_into().context("Too short symbol value")?,
-        ))
-        .context("Length overflow")?;
-        (ptr, len)
-    } else {
-        let (ptr, len) = value_slice[..8].split_at(4);
-        let ptr = usize::try_from(u32::from_le_bytes(
-            ptr.try_into().context("Too short symbol value")?,
-        ))
-        .context("Pointer overflow")?;
-        let len = usize::try_from(u32::from_le_bytes(
-            len.try_into().context("Too short symbol value")?,
-        ))
-        .context("Length overflow")?;
-        (ptr, len)
-    };
-    let chunk = &full_library_content[ptr - shift..ptr - shift + len];
-    serde_json::from_slice(chunk).with_context(|| {
-        format!(
-            "Failed to parse introspection chunk: '{}'",
-            String::from_utf8_lossy(chunk)
-        )
+fn deserialize_prefix(chunk: &[u8]) -> Result<Chunk> {
+    Chunk::deserialize(&mut Deserializer::from_slice(chunk)).with_context(|| {
+        // We take the first valid utf8 bytes, it's quite likely to be the actual chunk
+        // We use a 4096 upper bound for security
+        let chunk = str::from_utf8(&chunk[..4096])
```

Comment:
I guess this also breaks if the annotation is greater than 4096 chars?

Should we potentially store each static as a `#[repr(C)]` struct of `u32` size and then the bytes, so that we still have a length?